mongo/database.go

```diff
@@ -725,6 +725,18 @@ func (db *Database) createCollectionOperation(name string, opts ...*options.Crea
 			doc = bsoncore.AppendStringElement(doc, "granularity", *cco.TimeSeriesOptions.Granularity)
 		}
 
+		if cco.TimeSeriesOptions.BucketMaxSpanSeconds != nil {
```

Comment:
There are a few conditions for these new fields on the server ticket [SERVER-67598](https://jira.mongodb.org/browse/SERVER-67598) :

> We will enforce that 'bucketRoundingSeconds' is equal to 'bucketMaxSpanSeconds', and that both are strictly positive. If a user sets one value, they must set both. If a user sets these values, they cannot also set 'granularity'.

Should we enforce this in the library or rely on server errors?

Reply:
Considering the example on the server ticket:
> {timeseries: {timeField: "time", granularity: "seconds", bucketRoundingSeconds: 60, bucketMaxSpanSeconds: 3600}} // this is not an expected use case of the parameters but is allowed since the bucketMaxSpanSeconds and bucketRoundingSeconds are the default values corresponding to the 'seconds' granularity
I think it is easier to rely on server errors. Especially, it will be a runtime error either way.